TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/HttpJsonTransformingConsumer.java

```diff
@@ -90,14 +90,10 @@ private HttpRequestDecoder getHttpRequestDecoderHandler() {
 
     @Override
     public DiagnosticTrackableCompletableFuture<String, Void> consumeBytes(ByteBuf nextRequestPacket) {
-        chunks.add(nextRequestPacket.duplicate().readerIndex(0).retain());
+        chunks.add(nextRequestPacket.retainedDuplicate());
```

Comment:
@gregschohn, please verify if this behavior change is correct. I'm not sure why we were resetting the readerIndex before

Reply:
That's the impression I got too, will keep this as retainedDuplicate()